apps/designer/app/designer/features/style-panel/sections/typography/typography.tsx

```diff
@@ -1,19 +1,318 @@
+  const { sectionStyle } = props;
+
+  return (
+    <Grid
+      css={{
+        gridTemplateColumns: "1fr 1fr 1fr",
+        gap: "$spacing$5",
+      }}
+    >
+      <Grid css={{ gridTemplateColumns: "auto", gap: "$spacing$3" }}>
+        <PropertyName property="fontSize" label="Size" />
+        <TextControl {...sectionStyle.fontSize} />
+      </Grid>
+      <Grid css={{ gridTemplateColumns: "auto", gap: "$spacing$3" }}>
+        <PropertyName property="lineHeight" label="Height" />
+        <TextControl {...sectionStyle.lineHeight} />
+      </Grid>
+      <Grid css={{ gridTemplateColumns: "auto", gap: "$spacing$3" }}>
+        <PropertyName property="letterSpacing" label="Spacing" />
+        <TextControl {...sectionStyle.letterSpacing} />
+      </Grid>
+    </Grid>
+  );
+};
+
+export const TypographySectionAdvanced = (props: RenderCategoryProps) => {
+  const { setProperty, currentStyle } = props;
+  const setTextAlign = setProperty("textAlign");
+  const setTextDecorationLine = setProperty("textDecorationLine");
+  const setTextTransform = setProperty("textTransform");
+  const setFontStyle = setProperty("fontStyle");
+
+  return (
+    <Grid
+      css={{
+        gap: "$spacing$5",
+      }}
+    >
+      <Grid
+        css={{
+          gridTemplateColumns: "1fr 1fr",
+          gap: "$spacing$9",
+        }}
+      >
+        <ToggleGroupControl
+          property="textAlign"
+          onValueChange={(value) => setTextAlign(value)}
+          value={String(getTextAlign(String(currentStyle.textAlign?.value)))}
+          items={[
+            {
+              child: <TextAlignLeftIcon />,
+              label: "align: left",
+              value: "start",
+            },
+            {
+              child: <TextAlignCenterIcon />,
+              label: "align: center",
+              value: "center",
+            },
+            {
+              child: <TextAlignRightIcon />,
+              label: "align: right",
+              value: "end",
+            },
+            {
+              child: <TextAlignJustifyIcon />,
+              label: "align: justify",
+              value: "justify",
+            },
+          ]}
+        />
+        <ToggleGroupControl
+          property="textDecorationLine"
+          onValueChange={(value) => setTextDecorationLine(value)}
+          value={String(currentStyle.textDecorationLine?.value)}
+          items={[
+            {
+              child: <CrossSmallIcon />,
+              label: "None",
+              value: "none",
+            },
+            {
+              child: <TextUnderlineIcon />,
+              label: "Underline",
+              value: "underline",
+            },
+            {
+              child: <TextStrikethroughIcon />,
+              label: "Line through",
+              value: "line-through",
+            },
+          ]}
+        />
+      </Grid>
+      <Grid
+        css={{
+          gridTemplateColumns: "1fr 1fr auto",
```

Comment:
we might want to add this as a prop and allow passing a string, so that we don't need to use CSS overrides at all for this